app/src/components/evals/EvalForm/EvalForm.tsx

```diff
@@ -0,0 +1,192 @@
+import { Input, VStack, Text, Checkbox, HStack } from "@chakra-ui/react";
+import { ExternalModel } from "@prisma/client";
+import { useMemo } from "react";
+import AutoResizeTextArea from "~/components/AutoResizeTextArea";
+import InfoCircle from "~/components/InfoCircle";
+import InputDropdown from "~/components/InputDropdown";
+import frontendModelProvider from "~/modelProviders/openai-ChatCompletion/frontend";
+import { getOutputTitle } from "~/server/utils/getOutputTitle";
+import { ORIGINAL_MODEL_ID } from "~/types/dbColumns.types";
+import { predefinedEvaluationModels } from "~/utils/externalModels/evaluationModels";
+import { useDataset, useNodeEntries } from "~/utils/hooks";
+
+interface EvalNameProps {
+  datasetId: string;
+  nameState: {
+    name: string;
+    setName: React.Dispatch<React.SetStateAction<string>>;
+  };
+  modelState: {
+    includedModelIds: string[];
+    setIncludedModelIds: React.Dispatch<React.SetStateAction<string[]>>;
+  };
+  rowsState: {
+    numRows: number;
+    setNumRows: React.Dispatch<React.SetStateAction<number>>;
+    numComparisons: number;
+  };
+  evaluationModelState: {
+    selectedEvaluationModel: ExternalModel;
+    setSelectedEvaluationModel: React.Dispatch<React.SetStateAction<ExternalModel>>;
+  };
+  instructionsState: {
+    instructions: string;
+    setInstructions: React.Dispatch<React.SetStateAction<string>>;
+  };
+}
+
+const EvalForm: React.FC<EvalNameProps> = ({
+  datasetId,
+  nameState,
```

Comment:
Nit: I would probably deconstruct the state props right here in the function signature. I think you’ll have fewer lines of code overall, and I don’t think you gain much from doing it down in the function body.

Reply:
Let me try!